### ISSUE TITLE
[FEAT]: UX: Models Design Guidelines (update and align)

### ISSUE THREAD

**Issue by ngraham76** [DARPA-ASKEM/terarium#2249]
I have created a style guide for the model diagrams in Terarium to provide some guidance and consistency in how our models are designed. This guide covers basic and stratified petri net models and templates as well as templates used in the editing and coupling models operators. Please let me know if you have any questions, if you feel the guidelines are missing any important specification details or have any other feedback. Thanks!

Link to the Figma file: https://www.figma.com/file/yCstHNeLLPx0TzZC9uWXbl/Terarium---November-2023?type=design&node-id=2%3A99669&mode=design&t=K96TnWzzS6c2hHz1-1

Model sizes are relative to how and where they are displayed in the application (ie: operator previews, component cards, etc). The purpose of these dimensions is to provide a base set of specs to ensure size and scale consistency. Note: the font used to display variable node and connection labels has been change, it is now "**Latin Modern Math**".

**Basic Model Diagrams**
<img width="282" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/22ff25fa-2bb6-4d8d-ae58-0eb4a70a5a80">
<img width="789" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/41bf5134-983e-4730-be5d-9f50c226dd72">

**Stratified Model Diagrams**
<img width="1077" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/a68fbf5c-e03f-4251-94fa-7c316acd6377">
<img width="992" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/01c7fef7-2962-484f-b9ab-b1b5142b6eb9">
<img width="571" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/7205547d-7c84-4231-ac16-9c9be0ba2ed2">
<img width="595" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/9065bc08-a2ac-41d0-9fe1-531fe6983eb7">

**Model Templates**
Model templates are used in the Edit Model and Couple Models operators.
<img width="866" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/3c03b699-fc2e-4b65-8801-b022ca0f066c">
<img width="716" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/09fad8ad-9e7e-417e-a27b-7d480c5dda3f">
<img width="775" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/376a2d9d-d471-4752-bac4-02a839572610">
<img width="881" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/01ef2f8d-083a-4f56-b760-7d47143fb4a3">
<img width="1228" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/d3d0d332-312c-430b-9fea-2daf862bca8d">

**Comment by Tom-Szendrey** [comment]
Under **Basic Model Diagram**
Is this giving us the diameter of a node? Currently this will vary a lot due to our ability to zoom in and out is this meant for default?
What happens if the model is very complex? Do we start by default with a zoomed out picture to show the structure of the model or do we keep this size of the node and cut off whatever cannot fit 

<img width="88" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/17088680/aa5df863-30f0-4a08-975d-aa1f6b37b2df">

**Comment by Tom-Szendrey** [comment]
Does the colour pallet loop?

**Comment by Tom-Szendrey** [comment]
Is this legend here within the model diagram itself?
Will this be static (as in how does this legend respond to moving and zooming?)

<img width="256" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/17088680/25bd8d90-42d0-41ba-bced-3a7eb851ca7c">

**Comment by ngraham76** [comment]
> Does the colour pallet loop?

Yup, I should have mentioned that the palette reverses once we hit the darkest colour in the VIRDIS range:

<img width="595" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/36a1fedb-4a64-49f2-9bfb-28efbc6063c5">

**Comment by ngraham76** [comment]
> Is this legend here within the model diagram itself? Will this be static (as in how does this legend respond to moving and zooming?)
> 
> <img alt="image" width="256" src="https://user-images.githubusercontent.com/17088680/283506209-25bd8d90-42d0-41ba-bced-3a7eb851ca7c.png">

I imagine that the legend would be static on a canvas where the model diagram is able to be magnified and panned, something like this:

<img width="718" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/141876103/ffb3f416-de6c-4733-9046-c2e491018d66">

**Comment by ngraham76** [comment]
> Under **Basic Model Diagram** This is giving us the ratio between a node and transition lines correct? Currently this size itself will vary a lot due to our ability to zoom in and out is this meant for default
> 
> What happens if the model is very complex? Do we start by default with a zoomed out picture to show the structure of the model or do we keep this size of the node and cut off whatever cannot fit
> 
> <img alt="image" width="88" src="https://user-images.githubusercontent.com/17088680/283504316-aa5df863-30f0-4a08-975d-aa1f6b37b2df.png">

Yup, this was just a starting point to set up our sizes and scaling... but since it's re-sizable this goes out the window. I'll need to re-do this with "ems". And yes, if a model is larger and more complex we scale it down in size and fit into the preview or model operator card pane.

```
Em Unit: The em unit is based on the font size of the parent element.
For example, if the font size of a parent element is set to 16 pixels, then 1em is equal to 16 pixels.
```

**Comment by liunelson** [comment]
@ngraham76 Thanks for the changes that you suggested!
